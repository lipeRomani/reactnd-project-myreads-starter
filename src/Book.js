import React, {Component} from 'react';
import PropTypes from 'prop-types';

class Book extends Component {
    
    handleChange = e => {
        const shelfId = e.target.value;
        this.props.changeShelf(this.props.book, shelfId);
    }

    render() {
        return (
            <div className="book">
                <div className="book-top">
                    <div
                        className="book-cover"
                        style={{
                        width: 128,
                        height: 193,
                        backgroundImage: `url('${this.props.book.imageLinks ? this.props.book.imageLinks.smallThumbnail : ""}')`
                    }}></div>
                    <div className="book-shelf-changer" style={(this.props.book.shelf && this.props.book.shelf !== 'none') ? {backgroundColor : 'orange'} : {}}>
                        <select value={this.props.book.shelf ? this.props.book.shelf : 'none' } onChange={this.handleChange}>
                            <option value="none" disabled>Move to...</option>
                            <option value="currentlyReading">Currently Reading</option>
                            <option value="wantToRead">Want to Read</option>
                            <option value="read">Read</option>
                            <option value="none">None</option>
                        </select>
                    </div>
                </div>
                <div className="book-title">{this.props.book.title}</div>
                <div className="book-authors">{this.props.book.authors ? this.props.book.authors.join(", ") : ""}</div>
            </div>
        );
    }
}

Book.propTypes = {
    book : PropTypes.object.isRequired,
    changeShelf : PropTypes.func.isRequired
}

export default Book;
