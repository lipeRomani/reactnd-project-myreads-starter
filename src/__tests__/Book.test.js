import React from 'react';
import {shallow, render, mount} from 'enzyme';
import Book from '../Book';

it ('Test if componente render without error', () => {
    const book = {};
    const changeShelf = jest.fn();
    const wrapper = shallow(<Book book={book} changeShelf={changeShelf} />);
    expect(wrapper).toMatchSnapshot();
});

it ("Test if handle change is calling on change status", () => {
    const book = {};
    const changeShelf = jest.fn();

    const wrapper = mount(<Book book={book} changeShelf={changeShelf} />);
    wrapper.find('select').simulate('change');
    expect(changeShelf).toHaveBeenCalledTimes(1);
}); 

it ("Test book without authors", () => {
    const book = {
        authors : []
    };
    const changeShelf = jest.fn();
    const wrapper = mount(<Book book={book} changeShelf={changeShelf} />);
    expect(wrapper.find(".book-authors").text()).toEqual("");
});

it ("Test book with one author", () => {
    const author = "R. R. Martin";
    const book = {
        authors : [author]
    };
    const changeShelf = jest.fn();
    const wrapper = mount(<Book book={book} changeShelf={changeShelf} />);
    expect(wrapper.find(".book-authors").text()).toEqual(author);
});

it ("Test book with two authors", () => {
    const author1 = "R. R. Martin";
    const author2 = "Suzanne Collins";

    const book = {
        authors : [author1, author2]
    };
    const changeShelf = jest.fn();
    const wrapper = mount(<Book book={book} changeShelf={changeShelf} />);
    expect(wrapper.find(".book-authors").text()).toEqual(`${author1}, ${author2}`);
});

it ("test the color of the menu if the book is on the shelf", () => {
    const book = {
        shelf : "wantToRead"
    };
    const changeShelf = jest.fn();
    const wrapper = mount(<Book book={book} changeShelf={changeShelf} />);
    expect(wrapper.find(".book-shelf-changer").props().style.backgroundColor).toEqual("orange");
});

it ("test the color of the menu if the book is off the shelf", () => {
    const book = {};
    const changeShelf = jest.fn();
    const wrapper = mount(<Book book={book} changeShelf={changeShelf} />);
    expect(wrapper.find(".book-shelf-changer").props().style.backgroundColor).toEqual(undefined);
}); 