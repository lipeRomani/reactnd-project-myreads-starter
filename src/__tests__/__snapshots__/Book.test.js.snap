exports[`test Test if componente render without error 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="book">
    <div
        className="book-top">
        <div
            className="book-cover"
            style={
                Object {
                    "backgroundImage": "url(\'\')",
                    "height": 193,
                    "width": 128,
                  }
            } />
        <div
            className="book-shelf-changer"
            style={Object {}}>
            <select
                onChange={[Function]}
                value="none">
                <option
                    disabled={true}
                    value="none">
                    Move to...
                </option>
                <option
                    value="currentlyReading">
                    Currently Reading
                </option>
                <option
                    value="wantToRead">
                    Want to Read
                </option>
                <option
                    value="read">
                    Read
                </option>
                <option
                    value="none">
                    None
                </option>
            </select>
        </div>
    </div>
    <div
        className="book-title" />
    <div
        className="book-authors" />
</div>,
  "nodes": Array [
    <div
      className="book">
      <div
            className="book-top">
            <div
                  className="book-cover"
                  style={
                        Object {
                              "backgroundImage": "url(\'\')",
                              "height": 193,
                              "width": 128,
                            }
                  } />
            <div
                  className="book-shelf-changer"
                  style={Object {}}>
                  <select
                        onChange={[Function]}
                        value="none">
                        <option
                              disabled={true}
                              value="none">
                              Move to...
                        </option>
                        <option
                              value="currentlyReading">
                              Currently Reading
                        </option>
                        <option
                              value="wantToRead">
                              Want to Read
                        </option>
                        <option
                              value="read">
                              Read
                        </option>
                        <option
                              value="none">
                              None
                        </option>
                  </select>
            </div>
      </div>
      <div
            className="book-title" />
      <div
            className="book-authors" />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Book
        book={Object {}}
        changeShelf={[Function]} />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Book {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleChange": [Function],
        "props": Object {
          "book": Object {},
          "changeShelf": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="book">
          <div
                    className="book-top">
                    <div
                              className="book-cover"
                              style={
                                        Object {
                                                  "backgroundImage": "url(\'\')",
                                                  "height": 193,
                                                  "width": 128,
                                                }
                              } />
                    <div
                              className="book-shelf-changer"
                              style={Object {}}>
                              <select
                                        onChange={[Function]}
                                        value="none">
                                        <option
                                                  disabled={true}
                                                  value="none">
                                                  Move to...
                                        </option>
                                        <option
                                                  value="currentlyReading">
                                                  Currently Reading
                                        </option>
                                        <option
                                                  value="wantToRead">
                                                  Want to Read
                                        </option>
                                        <option
                                                  value="read">
                                                  Read
                                        </option>
                                        <option
                                                  value="none">
                                                  None
                                        </option>
                              </select>
                    </div>
          </div>
          <div
                    className="book-title" />
          <div
                    className="book-authors" />
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="book">
          <div
                    className="book-top">
                    <div
                              className="book-cover"
                              style={
                                        Object {
                                                  "backgroundImage": "url(\'\')",
                                                  "height": 193,
                                                  "width": 128,
                                                }
                              } />
                    <div
                              className="book-shelf-changer"
                              style={Object {}}>
                              <select
                                        onChange={[Function]}
                                        value="none">
                                        <option
                                                  disabled={true}
                                                  value="none">
                                                  Move to...
                                        </option>
                                        <option
                                                  value="currentlyReading">
                                                  Currently Reading
                                        </option>
                                        <option
                                                  value="wantToRead">
                                                  Want to Read
                                        </option>
                                        <option
                                                  value="read">
                                                  Read
                                        </option>
                                        <option
                                                  value="none">
                                                  None
                                        </option>
                              </select>
                    </div>
          </div>
          <div
                    className="book-title" />
          <div
                    className="book-authors" />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Book
    book={Object {}}
    changeShelf={[Function]} />,
}
`;
